sklearn/cluster/tests/test_optics.py

```diff
@@ -95,27 +95,18 @@ def test_dbscan_optics_parity():
     op = OPTICS(min_samples=10).fit(X)
     core_optics, labels_optics = op.extract_dbscan(0.3)
 
-    op_h = metrics.homogeneity_score(labels_true, labels_optics)
-    op_c = metrics.completeness_score(labels_true, labels_optics)
-    op_V = metrics.v_measure_score(labels_true, labels_optics)
-    op_m = metrics.adjusted_mutual_info_score(labels_true, labels_optics)
-    op_s = metrics.silhouette_score(X, labels_optics)
-
-    op_metrics = np.array([op_h, op_c, op_V, op_m, op_s])
-
-    # calculate dbscan labels and cluster metrics
+    # calculate dbscan labels
     db = DBSCAN(eps=0.3, min_samples=10).fit(X)
 
-    db_h = metrics.homogeneity_score(labels_true, db.labels_)
-    db_c = metrics.completeness_score(labels_true, db.labels_)
-    db_V = metrics.v_measure_score(labels_true, db.labels_)
-    db_m = metrics.adjusted_mutual_info_score(labels_true, db.labels_)
-    db_s = metrics.silhouette_score(X, db.labels_)
+    confusion_m = contingency_matrix(db.labels_, labels_optics)
+    agree = np.max(confusion_m, axis=1)
+    disagree = np.sum(confusion_m) - np.sum(agree)
```

Comment:
The number of agreements can be over-stated using this calculation.

```
[[5],
 [5],
 [5],
 [5]]
```

will be regarded as a perfect match to

```
[[20]]
```

You can fix this by using the max-sum assignment... or just by checking it is valid for axis=1 as well as axis=0. (Or `agree = min((... axis=1), (... axis=0))`)

Reply:
I'm not sure if I understand... `np.sum(confusion_m)` is just a stand in for `len(labels)`, which I can use instead if you prefer. I don't know what you mean by max-sum assignment, but `axis=0` and `axis=1` for the max call do give identical results; taking the min of both is perhaps a bit repetitive, but doesn't hurt, so I've changed it.